### ISSUE TITLE
JDI Collect Variables Hangs

### ISSUE THREAD

**Issue by RobertBColton** [enigma-dev/enigma-dev#1746]
This occurs if you try to build TKG's Key to Success game using emake. The problem is in `collect_variables` where JDI will infinite loop and hang. The problem is not in emake and this is also related to the lgmplugin.

Issue is explained here with steps to reproduce without TKG's full game:
https://github.com/enigma-dev/lgmplugin/issues/54

This issue is blocking the following fix to the plugin:
https://github.com/enigma-dev/lgmplugin/pull/71

This is the EDL generated from the Actions which JDI hangs parsing:
![Action EDL](https://user-images.githubusercontent.com/3212801/59398583-bd119880-8d5e-11e9-96ab-422695613072.png)

**Comment by RobertBColton** [comment]
I've gone and gathered some info from `collect_variables` now.

```
var__if__=0;{}__if__=action_if(0);if((__if__));
tttnnnnnn=0;{}nnnnnn=nnnnnnnnn(0);ss((nnnnnn));
```

**Comment by RobertBColton** [comment]
The reason the above happens is because prior to that, when it hits the first semicolon, it skips over the left brace and goes directly to handling the right brace.
https://github.com/enigma-dev/enigma-dev/blob/1ab91aefefe8fd7019ffbab876448b7bf65463b0/CompilerSource/parser/collect_variables.cpp#L157

**Comment by RobertBColton** [comment]
@JoshDreamland where should I go from here? Incrementing the position twice is clearly wrong, but I don't know if I should replace that increment with a continue or not.

**Comment by JoshDreamland** [comment]
Delete it and put the continue at the end of that if block.

**Comment by RobertBColton** [comment]
@JoshDreamland Alright, #1747 is up for review. I can't actually get it to pass the CI tests with the continue, so it just has the double increment commented. That works with my proposed plugin fixes and Sam's Key to Success now builds from emake. I went and regression tested all the games, so you know I looked hard for any side effects. Cheers!

**Comment by RobertBColton** [comment]
I later discovered where you wanted the `continue` statement was the same place I thought. I added it before merging and TKG's Key to Success still builds from emake!

**Comment by RobertBColton** [comment]
Closing as resolved by #1747.
